### PR TITLE
Options printing

### LINKED ISSUE #658: default `results` directory
If you install via pip, the default source tree isn't there.  At the moment, fitbenchmarking creates a directory named "results" in the directory the command is called from and stores output there.

It would be more helpful to the user if we created a directory somewhere (in the home directory?  `~/fitbenchmarking/results/`?) that was independent of the directory the command was run from.  Of course, this could be changed via an ini file.  Is it worth having this able to be changed via an environment variable too?

### LINKED ISSUE #663: Options printing
Adds the options to the log file and will print given to the terminal given a certain log level.

### PR DESCRIPTION
**Pull request by wathen**
#### Description of Work

Fixes #663 
Fixes #658 

Adds options printing to `DEBUG` log level and prints warning if `options.result_dir` is set to be the default.

#### Testing Instructions

1. Check options printing works for `DEBUG` log level
2. Check the wording of the warning for the results directory

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?

### CODE REVIEW

**Review by jfowkes — CHANGES_REQUESTED** [review]
Looks good apart from two issues:

1. For software not specified in FITTING, all solvers are printed, e.g:
```
scipy = CG
scipy_ls = lm-scipy-no-jac
	lm-scipy
	trf
	dogbox
```
even though I have only specified scipy as software and GS as minimiser.

2. Folder warning message should capitalise the and to, currently it reads:
```
WARNING: 
the FitBenchmarking results will be placed into the folder: 
   /Users/jari/Code/fitbenchmarking/fitbenchmarking_results
to change this alter the input options file.```

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 results/
+fitbenchmarking_results/
```
Comment (Anders-Markvardsen) [inline .gitignore:2]: with the renaming of result/ -> fitbenchmarking_results/, do we still want to ignore result/ also?

File: fitbenchmarking/cli/main.py [hunk 2]
```diff
@@ -142,11 +154,17 @@ def run(problem_sets, options_file='', debug=False):
 
         # resets options to original values
         options.reset()
+    if os.path.basename(options.results_dir) == \
+            options.DEFAULT_PLOTTING['results_dir']:
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/cli/main.py:158]: Somewhat unrelated to this PR, but what is the reason that 'results_dir' is located within

```
    DEFAULT_PLOTTING = \
        {'make_plots': True,
         'colour_scale': [(1.1, "#fef0d9"),
                          (1.33, "#fdcc8a"),
                          (1.75, "#fc8d59"),
                          (3, "#e34a33"),
                          (np.inf, "#b30000")],
         'comparison_mode': 'both',
         'table_type': ['acc', 'runtime', 'compare', 'local_min'],
         'results_dir': 'results'}
```

Does PLOTTING in DEFAULT_PLOTTING refer to anything that fitbenchmarking returns to the user? Maybe discuss this later in todays' friday catch-up

**Review by jfowkes — APPROVED** [review]
Looks great and works as intended for me.

**Pull request merged**